backup/slack_service.py

```diff
@@ -1,13 +1,29 @@
 import os
 
-from slack import WebClient
-from dotenv import find_dotenv, load_dotenv
+import slack
 
 
-load_dotenv(find_dotenv())
+class FakeWebClient:
+    """Fake slack WebClient"""
+    def __init__(self, post_message):
+        self.post_message = post_message
 
-client = WebClient(token=os.getenv("SLACK_API_TOKEN"))
+    def chat_postMessage(self, **kwargs):
+        return self.post_message(**kwargs)
 
 
-def send_message(message):
-    return client.chat_postMessage(channel="C0J8M5QMN", text=message)
+def make_slack_client(chat_post_message=None):
```

Comment:
Moving the logic of the `make_slack_client` into the `__init__` leads to tight coupling between the SlackService class and the client used for sending messages. That's why I create the clients outside and inject it into the class when instantiating it. That way I can pass in fake clients during test and the real clients in prod.

Reply:
Ideally, we'd need a fake client when testing. To do that, a more appropriate method is to mock it using `unittest.mock` and go on with our assertions.

The client is only needed by the SlackService class, we'd ideally never make use of it outside the class, hence the need for it to be part of the constructor of the class.
Infact, looking at this, we can export an instantiated class and keep using that instance to call multiple slack endpoints instead of always having to recreate another client, if we need to access another service on slack.